integration_tests/tests/utils/base_node_client.rs

```diff
@@ -0,0 +1,111 @@
+//  Copyright 2021. The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use std::{convert::TryInto, net::SocketAddr};
+
+use async_trait::async_trait;
+use log::trace;
+use tari_app_grpc::tari_rpc::{self as grpc, ConsensusConstants, GetShardKeyRequest};
+use tari_base_node_grpc_client::BaseNodeGrpcClient;
+use tari_common_types::types::{FixedHash, PublicKey};
+use tari_comms::types::CommsPublicKey;
+use tari_core::{blocks::BlockHeader, transactions::transaction_components::CodeTemplateRegistration};
+use tari_crypto::tari_utilities::ByteArray;
+use tari_validator_node::{error::GrpcBaseNodeError, types::BaseLayerMetadata};
+
+use crate::utils::base_node::BaseNodeClient;
+
+const LOG_TARGET: &str = "tari::validator_node::app";
+
+type Client = BaseNodeGrpcClient<tonic::transport::Channel>;
+
+#[derive(Clone)]
+pub struct GrpcBaseNodeClient {
+    endpoint: SocketAddr,
+    client: Option<Client>,
+}
+
+impl GrpcBaseNodeClient {
```

Comment:
I think it's going to be easier to use the rust GRPC client in `clients/rust/base_node_grpc_client` otherwise we're going to be duplicating a lot of code